src/peergos/server/tests/simulation/Simulator.java

```diff
@@ -515,16 +543,26 @@ private boolean verifySharingPermissions(String user, Path path) {
                         if (isVerified) {
                             try {
                                 // can overwrite?
-                                fs.write(path, new byte[]{read[0]});
-                            } catch (Exception ex) {
-                                LOG.log(Level.SEVERE, "User " + sharee + " could not write  shared-path " + path + "!", ex);
-                                isVerified = false;
+                                fs.write(path, read == null ? null : new byte[]{read[0]});
+                            } catch (Throwable ex) {
+                                //reload user and try again
+                                fs = fileSystems.getTestFileSystem(sharee, true);
+                                try {
+                                    fs.write(path, read == null ? null : new byte[]{read[0]});
```

Comment:
why is this necessary?

Reply:
That is the question. I don't know the answer at this stage. I am investigating. Making this change was sufficient for verify() to succeed.

Reply:
Maybe check it's not using a caching mutable pointers?

Reply:
grant code did not take into account any previous call to grant